### PR TITLE
feat(documents-v2): New documents client

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

* New documents client
* Add new resolver to documents api domain for possible usage

## Why

Old client is way out of date and is proving a nuisance for the logs. 

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by robertaandersen — APPROVED** [review]
LGTM

File: libs/clients/documents-v2/src/lib/documentsClientV2.service.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        ? CustomersListDocumentsOrderEnum[input.order]
+        : undefined,
+      sortBy: input.sortBy
+        ? CustomersListDocumentsSortByEnum[input.sortBy]
+        : undefined,
+    })
+
+    if (!documents.totalCount) {
+      //throw maybe?
+      return null
```
Comment (thordurhhh) [inline libs/clients/documents-v2/src/lib/documentsClientV2.service.ts:36]: I think returning null would be ideal instead of throwing here.

Comment (thorkellmani) [reply libs/clients/documents-v2/src/lib/documentsClientV2.service.ts:36]: ok so leave it as is

**Review by thordurhhh — APPROVED** [review]
Love it 😻

File: libs/clients/documents-v2/src/lib/documentsClientV2.service.ts (hunk 1 above)
Comment (saevarma) [reply libs/clients/documents-v2/src/lib/documentsClientV2.service.ts:36]: ~ Then remove the comment?

**Review by saevarma — APPROVED** [review]
Core files LGTM

File: libs/clients/documents-v2/src/lib/documentsClientV2.service.ts (hunk 1 above)
Comment (thorkellmani) [reply libs/clients/documents-v2/src/lib/documentsClientV2.service.ts:36]: will do

**Pull request merged**